bundles/framework/coordinatetool/plugin/CoordinatePluginHandler.js

```diff
@@ -0,0 +1,625 @@
+        data = data || this.getMapXY();
+
+        let fromProj = this.state.selectedProjection;
+        let toProj = this.mapModule.getProjection();
+        const successCb = (data) => {
+            if (showMarker) {
+                this.addMarker(data, markerMessageData);
+            }
+
+            if (showMarker || centerMap) {
+                this.centerMap(data);
+            }
+
+            if (!centerMap) {
+                this.updateLonLat(data, true);
+            }
+        };
+        const errorCb = () => {
+            this.setLoading(false);
+        };
+
+        this.setLoading(true);
+
+        if (swapProjections) {
+            fromProj = this.mapModule.getProjection();
+            toProj = this.state.selectedProjection;
+        }
+
+        this.coordinateTransformationExtension.getTransformedCoordinatesFromServer(data, fromProj, toProj, successCb, errorCb);
+        this.setLoading(false);
+    }
+
+    updateReverseGeocode (data) {}
+
+    getEmergencyCallCoordinatesFromServer (data, cb) {
+        // get the transform from current data
+        const sourceProjection = this.mapModule.getProjection();
+
+        // If coordinates are not  EPSG:4326 then
+        // need to get 'EPSG:4326' coordinates from service
+        if (sourceProjection !== 'EPSG:4326') {
+            this.coordinateTransformationExtension.getTransformedCoordinatesFromServer(data, sourceProjection, 'EPSG:4326',
+                (responseDataTo4326) => {
+                    cb(this.formatEmergencyCallMessage(responseDataTo4326));
+                },
+                () => {});
+        }
+        // Else if coordinates are from 'EPSG:4326' then use these
+        else {
+            return this.formatEmergencyCallMessage(data);
+        }
+    }
+
+    getEmergencyCallInfo (data) {
+        const xy = data || this.getMapXY();
+
+        // update emergency if configured
+        if (this.config.showEmergencyCallMessage) {
+            // already in degrees, don't fetch again
+            if (this.allowDegrees() && this.mapModule.getProjection() === 'EPSG:4326') {
+                this.updateState({
+                    emergencyInfo: this.formatEmergencyCallMessage({
+                        'lonlat': {
+                            'lon': xy.lonlat.lon,
+                            'lat': xy.lonlat.lat
+                        }
+                    })
+                });
+            } else {
+                this.getEmergencyCallCoordinatesFromServer(data, (serverData) => {
+                    this.updateState({
+                        emergencyInfo: serverData
+                    });
+                });
+            }
+        } else {
+            this.updateState({
+                emergencyInfo: null
+            });
+        }
+        this.updatePopup();
+    }
+
+    formatEmergencyCallMessage (data) {
+        const degmin = this.coordinateTransformationExtension._formatDegrees(data.lonlat.lon, data.lonlat.lat, 'min');
```

Comment:
Could we use this from Oskari.util instead? https://github.com/oskariorg/oskari-frontend/blob/develop/src/util.js#L507